### PR TITLE
feat: THES-34: PaymentProcessingDialog dialog created

### PR DESCRIPTION
**Pull request by grmartin**
- Dialog UI Created
- Tests as well

### CODE REVIEW

File: src/payment/PaymentProcessingModal.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+});
+
+/**
+ * PaymentProcessingModal
+ */
+export const PaymentProcessingModal = ({
+  show,
+}) => {
+  const intl = useIntl();
+  const paymentStatusSelector = () => ({ status: 'testing' });
```
Comment (grmartin) [inline src/payment/PaymentProcessingModal.jsx:32]: for THEX-105 and such, this needs to be pulled from a data selector/reducer.

File: src/payment/PaymentProcessingModal.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from 'react';
+
+import {
+  ModalDialog, Spinner,
+} from '@edx/paragon';
+import { useSelector } from 'react-redux';
+import { useIntl, defineMessages } from '@edx/frontend-platform/i18n';
+import PropTypes from 'prop-types';
+
+// import { subscriptionStatusSelector } from '../data/status/selectors';
```
Comment (wdrussell2015) [inline src/payment/PaymentProcessingModal.jsx:10]: Meant to be paymentStatusSelector as a TODO? or just remove.

File: src/payment/PaymentProcessingModal.jsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+const messages = defineMessages({
```
Comment (wdrussell2015) [inline src/payment/PaymentProcessingModal.jsx:12]: nit: separate into PaymentProcessingModal.messages.jsx

**Review by wdrussell2015 — APPROVED** [review]
minor comments but LGTM!

File: src/payment/PaymentProcessingModal.jsx (hunk 2 above)
Comment (grmartin) [reply src/payment/PaymentProcessingModal.jsx:10]: Ill comment on it and leave a TODO.

File: src/payment/PaymentProcessingModal.jsx (hunk 3 above)
Comment (grmartin) [reply src/payment/PaymentProcessingModal.jsx:12]: Ill do this morning

File: src/payment/PaymentProcessingModal.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+export const PaymentProcessingModal = ({
+  show,
+}) => {
+  const intl = useIntl();
+  const paymentStatusSelector = () => ({ status: 'testing' });
+  const { status } = useSelector(paymentStatusSelector);
+  const [isOpen, setOpen] = useState(show);
+
+  useEffect(() => {
+    if (status === 'failed' || status === 'completed') {
```
[the remark this answers is not in the document]
Comment (grmartin) [reply src/payment/PaymentProcessingModal.jsx]: no idea, will PL

[the remark this answers is not in the document]
Comment (grmartin) [reply src/payment/PaymentProcessingModal.jsx]: Will rework today to do autoshow/hide. Removing the show option.

[the remark this answers is not in the document]
Comment (grmartin) [reply src/payment/PaymentProcessingModal.jsx]: this is complete

**Review by wdrussell2015 — APPROVED** [review]
Ship It!

**Pull request merged**